### ISSUE TITLE
.

### ISSUE THREAD

**Issue by olegatorizshayan** [olegatorizshayan/txt_reader#1]
// compile with
// g++ gui.cpp -o gui -lsfml-graphics -lsfml-window -lsfml-system

#include <thread>
#include <iostream>
#include <SFML/Graphics.hpp>
#include <vector>
#include <math.h>

bool menu = 0;
bool reading = 1;
bool nextPage = 0;
bool prevPage = 0;
std::vector<char> book;
int index = 0;
sf::Color textColor = sf::Color::Black;
sf::Color backgroundColor = sf::Color::White;
bool vozim = 0;
int mainWindow_left;
int mainWindow_top;
int mainWindow_width;
int mainWindow_height;

int READ() {
	if (FILE *fp = fopen("book.txt", "r"))
	{
		char buf[1024];
		while (size_t len = fread(buf, 1, sizeof(buf), fp))
			book.insert(book.end(), buf, buf + len);
		fclose(fp);
	} else return 0;
	return 1;
}

void text_color() {
	int window_width = 300;
	int window_height = 300;
    sf::RenderWindow window(sf::VideoMode(window_width, window_height), "TEXT");
    window.setPosition(sf::Vector2i(mainWindow_left+mainWindow_width,mainWindow_top));

	// каждый охотник желает знать где сидит фазан
	sf::Color rainbow[7] = {sf::Color(255,0,0),
	sf::Color(255,128,0),
	sf::Color::Yellow,
	sf::Color(0,255,0),
	sf::Color(0,191,255),
	sf::Color(0,0,255),
	sf::Color(90,0,157)};

	int radius = window_width/2;
	int x = window_width/2;
	int y = window_height/2;

	window.clear(backgroundColor);
		for(double i = 0.25; i<=radius; i+=0.25) {
			for(int j = 0; j < 7; j++) {
				sf::VertexArray lines(sf::Lines, 2);
				double angle = 360.0/7*j*0.0174533;
				lines[0].position = sf::Vector2f(x+cos(angle)*i,y+sin(angle)*i);
				angle+=360.0/7*0.0174533;
				lines[1].position = sf::Vector2f(x+cos(angle)*i,y+sin(angle)*i);
				lines[0].color = rainbow[j];
				lines[1].color = rainbow[(j+1)%7];
				if(i<10) {
					lines[0].color = sf::Color::Black;
					lines[1].color = sf::Color::Black;
				}
				window.draw(lines);
			}
		}
	sf::Image screenshot = window.capture();
	window.display();
	
	
    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
            if (event.type == sf::Event::MouseButtonPressed) {
            	vozim = true;
            }
            if (event.type == sf::Event::MouseButtonReleased) {
            	vozim = false;
            }
            if (vozim) {
            	int mx = sf::Mouse::getPosition(window).x;
            	int my = sf::Mouse::getPosition(window).y;
            	if(sqrt((mx-x)*(mx-x)+(my-y)*(my-y)) <= radius) {
            		sf::Color pixel = screenshot.getPixel(mx,my);
	            	textColor = pixel;
	            	std::cout << (int)pixel.r << " " << (int)pixel.g << " " << (int)pixel.b << std::endl;
            	}
            }
        }
    }
}

void background_color() {
	int window_width = 300;
	int window_height = 300;
    sf::RenderWindow window(sf::VideoMode(window_width, window_height), "BACKGROUND");
    window.setPosition(sf::Vector2i(mainWindow_left+mainWindow_width,mainWindow_top+mainWindow_height-window.getSize().y));
    

	// каждый охотник желает знать где сидит фазан
	sf::Color rainbow[7] = {sf::Color(255,0,0),
	sf::Color(255,128,0),
	sf::Color::Yellow,
	sf::Color(0,255,0),
	sf::Color(0,191,255),
	sf::Color(0,0,255),
	sf::Color(90,0,157)};

	int radius = window_width/2;
	int x = window_width/2;
	int y = window_height/2;

	window.clear(backgroundColor);
		for(double i = 0.25; i<=radius; i+=0.25) {
			for(int j = 0; j < 7; j++) {
				sf::VertexArray lines(sf::Lines, 2);
				double angle = 360.0/7*j*0.0174533;
				lines[0].position = sf::Vector2f(x+cos(angle)*i,y+sin(angle)*i);
				angle+=360.0/7*0.0174533;
				lines[1].position = sf::Vector2f(x+cos(angle)*i,y+sin(angle)*i);
				lines[0].color = rainbow[j];
				lines[1].color = rainbow[(j+1)%7];
				if(i<10) {
					lines[0].color = sf::Color::Black;
					lines[1].color = sf::Color::Black;
				}
				window.draw(lines);
			}
		}
	sf::Image screenshot = window.capture();
	window.display();
	
	
    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
            if (event.type == sf::Event::MouseButtonPressed) {
            	vozim = true;
            }
            if (event.type == sf::Event::MouseButtonReleased) {
            	vozim = false;
            }
            if (vozim) {
            	int mx = sf::Mouse::getPosition(window).x;
            	int my = sf::Mouse::getPosition(window).y;
            	if(sqrt((mx-x)*(mx-x)+(my-y)*(my-y)) <= radius) {
            		sf::Color pixel = screenshot.getPixel(mx,my);
	            	backgroundColor = pixel;
	            	std::cout << (int)pixel.r << " " << (int)pixel.g << " " << (int)pixel.b << std::endl;
            	}
            }
        }
    }
}
#include <SFML/Window.hpp>
int main()
{
	sf::String string = sf::Clipboard::getString();
	std::vector<char> text;
	for(int i = 0; ; i++)
		if(string[i]=='\0') break;
		else text.push_back((char)string[i]);

	
	int screen_width = sf::VideoMode::getDesktopMode().width;
	int screen_height = sf::VideoMode::getDesktopMode().height;

	//if(!READ()) {
		//std::cout << "can't read file!\n";
		//return 0;
	//}

	int window_width,window_height,button_height,button_width;
	
	window_width = sf::VideoMode::getDesktopMode().width;//600;
	window_height = sf::VideoMode::getDesktopMode().height;//700;
    sf::RenderWindow window(sf::VideoMode(window_width, window_height), "MAGIC BOOK");
    mainWindow_width = window_width;
    mainWindow_height = window_height;
    mainWindow_left = window.getPosition().x;
    mainWindow_top = window.getPosition().y;

	std::thread t(text_color);
	t.detach();
	std::thread t2(background_color);
	t2.detach();

	
    

    button_height = window_height/3;
    button_width = window_width;

    sf::CircleShape but1(1.f);
    but1.scale(button_width/2,button_height/2);
    but1.setFillColor(sf::Color::Green);
    but1.setPosition(0,(window_height-2*button_height)/3);

    sf::CircleShape but2(1.f);
    but2.scale(button_width/2,button_height/2);
    but2.setFillColor(sf::Color::Green);
    but2.setPosition(0,(window_height-2*button_height)/3*2+button_height);
    
    sf::Font font;
    font.loadFromFile("sansation.ttf");

    sf::Text label1;
    label1.setFont(font);
    label1.setString("READ");
    int label1_width,label1_height;
    label1_width = label1.getGlobalBounds().width;
    label1_height = label1.getGlobalBounds().height;
    label1.setPosition((window_width-label1_width)/2,(window_height-2*button_height)/3+(button_height-label1_height)/2);

    sf::Text label2;
    label2.setFont(font);
    label2.setString("SETTINGS");
    int label2_width,label2_height;
    label2_width = label2.getGlobalBounds().width;
    label2_height = label2.getGlobalBounds().height;
    label2.setPosition((window_width-label2_width)/2,(window_height-2*button_height)/3*2+button_height+(button_height-label2_height)/2);

    while(!window.hasFocus()) {
    	window.requestFocus();
    	std::cout << "ijdfafkpsopfaosjk okfdop askfop kasokf opask";
    
    }
    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
            if (reading && sf::Keyboard::isKeyPressed(sf::Keyboard::Right))
            	nextPage = 1;
            if (reading && sf::Keyboard::isKeyPressed(sf::Keyboard::Left))
            	prevPage = 1;
        }
        window.clear(backgroundColor);
        if(menu) {
	        window.draw(but1);
	        window.draw(but2);
	        window.draw(label1);
	        window.draw(label2);
	    }
	    if(reading) {
	    	int x,y,maxHeightInRow,otstupSleva,otstupSprava,otstupSverhu,otstupSnizu,mezhStrochInt,mezhBukvInt;
	    	mezhStrochInt = 35;
	    	mezhBukvInt = 5;
	    	otstupSleva = 50;
	    	otstupSprava = 50;
	    	otstupSnizu = 50;
	    	otstupSverhu = 0;
		    maxHeightInRow = 0;
		    sf::Text character;
		    character.setColor(textColor);
		    character.setFont(font);
	    	if(!prevPage) {
		    	x = otstupSleva;
		    	y = otstupSverhu;
		    	for(int i = index; i < book.size(); i++) {
		    		character.setString(book[i]);
		    		//std::cout << character.getGlobalBounds().height << std::endl;
		    		if(x+character.getGlobalBounds().width>window_width-otstupSprava || book[i]=='\n') {
		    			y += maxHeightInRow+mezhStrochInt;
		    			x = otstupSleva;
		    			maxHeightInRow = 0;
		    		}
		    		if(y+character.getGlobalBounds().height>window_height-otstupSnizu) {
		    			if(nextPage)
		    				index = i;
		    			break;
		    		}
		    		character.setPosition(x,y);
		    		window.draw(character);
		    		x += character.getGlobalBounds().width+mezhBukvInt;
		    		if(character.getGlobalBounds().height>maxHeightInRow)
		    			maxHeightInRow = character.getGlobalBounds().height;
		    	}
		    	nextPage = 0;
		    }
		    if(prevPage && index == 0) prevPage = 0;
		    if(prevPage) {
				int index_old = index;
			    index = 0;
			    x = otstupSleva;
			    y = otstupSverhu;
			    for(int i = 0; i < book.size(); i++) {
			    	character.setString(book[i]);
			    	if(x+character.getGlobalBounds().width>window_width-otstupSprava || book[i]=='\n') {
			    		y += maxHeightInRow+mezhStrochInt;
			    		x = otstupSleva;
			    		maxHeightInRow = 0;
			    	}
			    	if(y+character.getGlobalBounds().height>window_height-otstupSnizu) {
			    		if(i>=index_old) break;
			    		index = i;
			    		x = otstupSleva;
			    		y = otstupSverhu;
			    		maxHeightInRow = 0;
			    	}
			    	//character.setPosition(x,y);
			    	//window.draw(character);
			    	x += character.getGlobalBounds().width+mezhBukvInt;
			    	if(character.getGlobalBounds().height>maxHeightInRow)
			    		maxHeightInRow = character.getGlobalBounds().height;
			    }
		    	prevPage = 0;
		    }
	    }
        window.display();
    }

    return 0;
}
